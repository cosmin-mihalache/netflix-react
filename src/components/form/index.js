import React from 'react';
import {
  Container,
  Base,
  Title,
  Submit,
  Text,
  SmallText,
  Input,
  Link,
  Error,
} from './styles/form';

export default function Form({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Form.Base = function FormBase({ children, ...restProps }) {
  return <Base {...restProps}>{children}</Base>;
};

Form.Title = function FormTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

Form.Error = function FormError({ children, ...restProps }) {
  return <Error {...restProps}>{children}</Error>;
};

Form.Input = function FormInput({ children, ...restProps }) {
  return <Input {...restProps}>{children}</Input>;
};

Form.Submit = function FormSubmit({ children, ...restProps }) {
  return <Submit {...restProps}>{children}</Submit>;
};

Form.Link = function FormLink({ children, ...restProps }) {
  return <Link {...restProps}>{children}</Link>;
};

Form.Text = function FormText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Form.SmallText = function FormSmallText({ children, ...restProps }) {
  return <SmallText {...restProps}>{children}</SmallText>;
};
