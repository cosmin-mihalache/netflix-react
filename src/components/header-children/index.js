import React from 'react';
import { Container, Title, SubTitle } from './styles/header-children';

export default function HeaderChild({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

HeaderChild.Title = function HeaderChildTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

HeaderChild.SubTitle = function HeaderChildSubTitle({
  children,
  ...restProps
}) {
  return <SubTitle {...restProps}>{children}</SubTitle>;
};
