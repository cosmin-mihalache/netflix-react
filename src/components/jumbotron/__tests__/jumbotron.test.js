import React from 'react';
import { render } from '@testing-library/react';
import { Jumbotron } from '../../../components';
import jumbotronData from '../../../fixtures/jumbotron.json';

describe('<Jumbotron />', () => {
  it('renders the <Jumbotron /> with populated data', () => {
    const { container, getByText, getAllByAltText, getByTestId } = render(
      <Jumbotron.Container>
        {jumbotronData.map((item) => (
          <Jumbotron key={item.id}>
            <Jumbotron.Pane>
              <Jumbotron.Title>{item.title}</Jumbotron.Title>
              <Jumbotron.SubTitle>{item.subTitle}</Jumbotron.SubTitle>
            </Jumbotron.Pane>
            <Jumbotron.Pane>
              <Jumbotron.Image
                src={item.image}
                alt={item.alt}
                data-testid={`${item.id}-jumbotron`}
              />
            </Jumbotron.Pane>
          </Jumbotron>
        ))}
      </Jumbotron.Container>
    );

    expect(getByText('Enjoy on your TV.')).toBeTruthy();
    expect(getByTestId('1-jumbotron')).toBeTruthy();
    expect(
      getByText(
        'Watch on Smart TVs, Playstation, Xbox, Chromecast, Apple TV, Blu-ray players, and more.'
      )
    ).toBeTruthy();
    expect(getAllByAltText('Tiger King | Netflix')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
