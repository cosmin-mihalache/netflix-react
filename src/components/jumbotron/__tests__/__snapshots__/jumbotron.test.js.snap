// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Jumbotron /> renders the <Jumbotron /> with populated data 1`] = `
<section
  class="jumbotron__Container-sc-1gvy20t-3 eLoyvV"
>
  <div
    class="jumbotron__Item-sc-1gvy20t-2 iIxdmG"
  >
    <div
      class="jumbotron__Inner-sc-1gvy20t-0 iQNYla"
      direction="row"
    >
      <div
        class="jumbotron__Pane-sc-1gvy20t-1 hACuLn"
      >
        <h1
          class="jumbotron__Title-sc-1gvy20t-4 gyzSCg"
        >
          Enjoy on your TV.
        </h1>
        <h2
          class="jumbotron__SubTitle-sc-1gvy20t-5 bnJuoF"
        >
          Watch on Smart TVs, Playstation, Xbox, Chromecast, Apple TV, Blu-ray players, and more.
        </h2>
      </div>
      <div
        class="jumbotron__Pane-sc-1gvy20t-1 hACuLn"
      >
        <img
          alt="Tiger King | Netflix"
          class="jumbotron__Image-sc-1gvy20t-6 lbiJhz"
          data-testid="1-jumbotron"
          src="./images/misc/home-tv.jpg"
        />
      </div>
    </div>
  </div>
  <div
    class="jumbotron__Item-sc-1gvy20t-2 iIxdmG"
  >
    <div
      class="jumbotron__Inner-sc-1gvy20t-0 iQNYla"
      direction="row"
    >
      <div
        class="jumbotron__Pane-sc-1gvy20t-1 hACuLn"
      >
        <h1
          class="jumbotron__Title-sc-1gvy20t-4 gyzSCg"
        >
          Download your shows to watch offline.
        </h1>
        <h2
          class="jumbotron__SubTitle-sc-1gvy20t-5 bnJuoF"
        >
          Save your favorites easily and always have something to watch.
        </h2>
      </div>
      <div
        class="jumbotron__Pane-sc-1gvy20t-1 hACuLn"
      >
        <img
          alt="Tiger King | Netflix"
          class="jumbotron__Image-sc-1gvy20t-6 lbiJhz"
          data-testid="2-jumbotron"
          src="./images/misc/home-mobile.jpg"
        />
      </div>
    </div>
  </div>
  <div
    class="jumbotron__Item-sc-1gvy20t-2 iIxdmG"
  >
    <div
      class="jumbotron__Inner-sc-1gvy20t-0 iQNYla"
      direction="row"
    >
      <div
        class="jumbotron__Pane-sc-1gvy20t-1 hACuLn"
      >
        <h1
          class="jumbotron__Title-sc-1gvy20t-4 gyzSCg"
        >
          Watch everywhere.
        </h1>
        <h2
          class="jumbotron__SubTitle-sc-1gvy20t-5 bnJuoF"
        >
          Stream unlimited movies and TV shows on your phone, tablet, laptop, and TV without paying more.
        </h2>
      </div>
      <div
        class="jumbotron__Pane-sc-1gvy20t-1 hACuLn"
      >
        <img
          alt="Money Heist | Netflix"
          class="jumbotron__Image-sc-1gvy20t-6 lbiJhz"
          data-testid="3-jumbotron"
          src="./images/misc/home-imac.jpg"
        />
      </div>
    </div>
  </div>
  <div
    class="jumbotron__Item-sc-1gvy20t-2 iIxdmG"
  >
    <div
      class="jumbotron__Inner-sc-1gvy20t-0 iQNYla"
      direction="row"
    >
      <div
        class="jumbotron__Pane-sc-1gvy20t-1 hACuLn"
      >
        <h1
          class="jumbotron__Title-sc-1gvy20t-4 gyzSCg"
        >
          Get the Netflix Kids experience.
        </h1>
        <h2
          class="jumbotron__SubTitle-sc-1gvy20t-5 bnJuoF"
        >
          Kids enjoy their favorites in profiles made just for them, and parental controls ensure theyâ€™re watching the right things.
        </h2>
      </div>
      <div
        class="jumbotron__Pane-sc-1gvy20t-1 hACuLn"
      >
        <img
          alt="Netflix Kids"
          class="jumbotron__Image-sc-1gvy20t-6 lbiJhz"
          data-testid="4-jumbotron"
          src="./images/misc/netflix-kids.jpg"
        />
      </div>
    </div>
  </div>
</section>
`;
